clients/da-vinci-client/src/main/java/com/linkedin/davinci/consumer/VeniceAfterImageConsumerImpl.java

```diff
@@ -118,24 +118,40 @@ protected static void adjustSeekCheckPointsBasedOnHeartbeats(
       }
       PubSubPosition heartbeatTimestampPosition =
           consumerAdapter.getPositionByTimestamp(topicPartition, currentVersionTimestamp);
-      if (checkpoints.get(topicPartition.getPartitionNumber()) == null) {
-        LOGGER.warn("No EOP checkpoint found for partition: {}", topicPartition.getPartitionNumber());
+      if (checkpoints.get(topicPartition.getPartitionNumber()) == null && heartbeatTimestampPosition == null) {
+        LOGGER.warn(
+            "No EOP checkpoint OR heartbeat position found for partition: {} seeking to tail",
+            topicPartition.getPartitionNumber());
         checkpoints.put(
             topicPartition.getPartitionNumber(),
             new VeniceChangeCoordinate(
                 topicPartition.getPubSubTopic().getName(),
-                heartbeatTimestampPosition,
+                PubSubPosition.LATEST,
```

Comment:
Should we use EARLIEST to be safe?

Reply:
Good question.  I struggled with this a bit..... and I may change my mind again admittedly...  This should probably actually be configurable IMO, it's kind of like the off retention strategy in a Kafka Consumer.  Going to earliest makes sure there isn't any data gap, going to latest makes sure the client is overwhelmed with writes due to consuming the whole version.  We right now have more partners that want to be caught up vs. consuming everything all over again.  So I think what I'd like to do is put this to a later PR to make this a configurable behavior and to add some kind of metric for when this happens (as this should be an extreme scenario).

Reply:
Sounds reasonable. Thanks for the clarification